src/parser/ParsedQuery.h

```diff
@@ -357,11 +359,29 @@ class ParsedQuery {
     [[nodiscard]] auto& getSelectVariables()  {
       return std::get<std::vector<string>>(_varsOrAsterisk);
     }
+
+    void pushVariablesOrder(const string& variable) {
+      if(!(std::find(_variablesOrder.begin(),
+                    _variablesOrder.end(),
+                    variable) != _variablesOrder.end())) {
+        _variablesOrder.emplace_back(variable);
+      }
+    }
+
+
+    [[nodiscard]] std::list<string>& retrieveOrder() {
+      return _variablesOrder;
+    }
+
+    [[nodiscard]] std::list<string> retrieveOrder() const {
```

Comment:
call this `orderedVariablesFromQueryBody` and comment

`// Return the variables that were added by the `addVariableFromQueryBody` function. The return value does not contain duplicates.